cli/cmd/generate_azure.go

```diff
@@ -0,0 +1,680 @@
+
+	generateAzureTfCommand.PersistentFlags().BoolVar(
+		&GenerateAzureCommandState.ExistingStorageAccount, "existing_storage", false, "use existing storage account")
+	generateAzureTfCommand.PersistentFlags().StringVar(
+		&GenerateAzureCommandState.StorageAccountName,
+		"strg_account_name",
+		"",
+		"specify storage account name")
+	generateAzureTfCommand.PersistentFlags().StringVar(
+		&GenerateAzureCommandState.StorageAccountResourceGroup,
+		"strg_res_group",
+		"",
+		"specify storage resource group")
+	generateAzureTfCommand.PersistentFlags().StringVar(
+		&GenerateAzureCommandState.StorageLocation,
+		"location",
+		"",
+		"specify azure region where storage account logging resides")
+
+	generateAzureTfCommand.PersistentFlags().BoolVar(
+		&GenerateAzureCommandState.AllSubscriptions, "all_subs", false, "grant read access to ALL subscriptions within Tenant")
+
+	generateAzureTfCommand.PersistentFlags().StringSliceVar(
+		&GenerateAzureCommandState.SubscriptionIds,
+		"sub_ids",
+		[]string{},
+		`list of subscriptions to grant read access; format is id1,id2,id3`)
+
+	generateAzureTfCommand.PersistentFlags().StringVar(
+		&GenerateAzureCommandState.AdApplicationPassword,
+		"ad_pass",
+		"",
+		"password for an already existing active directory application")
+
+	generateAzureTfCommand.PersistentFlags().StringVar(
+		&GenerateAzureCommandState.AdApplicationId,
+		"ad_id",
+		"",
+		"id for an already existing active directory application")
+
+	generateAzureTfCommand.PersistentFlags().StringVar(
+		&GenerateAzureCommandState.AdServicePrincipalId,
+		"ad_pid",
+		"",
+		"principle id for an already existing active directory application")
+
+	generateAzureTfCommand.PersistentFlags().BoolVar(
+		&GenerateAzureCommandExtraState.TerraformApply, "terraform-apply", false, "run terraform apply for the generated hcl")
+
+}
+
+func promptAzureIntegrationNameQuestions(config *azure.GenerateAzureTfConfigurationArgs) error {
+
+	if err := SurveyMultipleQuestionWithValidation([]SurveyQuestionWithValidationArgs{
+		{
+			Prompt:   &survey.Input{Message: QuestionAzureConfigName, Default: config.ConfigIntegrationName},
+			Checks:   []*bool{&config.Config},
+			Required: true,
+			Response: &config.ConfigIntegrationName,
+		},
+		{
+			Prompt:   &survey.Input{Message: QuestionActivityLogName, Default: config.ActivityLogIntegrationName},
+			Checks:   []*bool{&config.ActivityLog},
+			Required: true,
+			Response: &config.ActivityLogIntegrationName,
+		},
+	}); err != nil {
+		return err
+	}
+	return nil
+}
+
+func promptAzureStorageAccountQuestions(config *azure.GenerateAzureTfConfigurationArgs) error {
+
+	if err := SurveyMultipleQuestionWithValidation([]SurveyQuestionWithValidationArgs{
+		{
+			Prompt:   &survey.Confirm{Message: QuestionUseExistingStorageAccount, Default: config.ExistingStorageAccount},
+			Response: &config.ExistingStorageAccount,
+		},
+		{
+			Prompt:   &survey.Input{Message: QuestionAwsRegion, Default: config.StorageLocation},
```

Comment:
Looks like this is asking an AWS question here

Reply:
Should we be asking for the location here? It seems like we should be asking for the region at the top level.
Or at least we should only be asking for the region/location if `config.ExistingStorageAccount` is false

Reply:
Think that this is out of date, need to see that latest code